lib/config/migrations.go

```diff
@@ -96,6 +99,43 @@ func (m migration) apply(cfg *Configuration) {
 	cfg.Version = m.targetVersion
 }
 
+func migrateToConfigV38(cfg *Configuration) {
+	// Renamed notification ID
+	for i := range cfg.Options.UnackedNotificationIDs {
+		if cfg.Options.UnackedNotificationIDs[i] == "authenticationUserAndPassword" {
+			cfg.Options.UnackedNotificationIDs[i] = "guiAuthentication"
+		}
+	}
+
+	// New required settings
+	defaultGuiCfg := structutil.WithDefaults(GUIConfiguration{})
+	if cfg.GUI.WebauthnRpId == "" {
+		cfg.GUI.WebauthnRpId = defaultGuiCfg.WebauthnRpId
+	}
+	if len(cfg.GUI.RawWebauthnOrigins) == 0 {
+		_, port, err := net.SplitHostPort(cfg.GUI.RawAddress)
```

Comment:
I just noticed this won't work when generating an initial configuration.  No idea currently how we can work around that.

Reply:
I'll look at the main issue here after we've resolved https://github.com/emlun/syncthing/pull/7, but for now:

> why do I need to set a username actually?

It's really just so that we have something sensible to set as the [`user.name`](https://www.w3.org/TR/webauthn-3/#dom-publickeycredentialentity-name) (and [`user.displayName`](https://www.w3.org/TR/webauthn-3/#dom-publickeycredentialuserentity-displayname)) option when creating the credential, rather than something generic like "Syncthing" or maybe "Syncthing @ <Device-Name>". That happens [here](https://github.com/syncthing/syncthing/blob/928d6ff774bf71e387dcd0f07033e8f977e58197/lib/api/api_auth_webauthn.go#L137-L142) in the methods required by the `go-webauthn/User` interface.

These options' only role is to be shown in credential picker/management UIs, so I suppose we _could_ use something simple if we want to get rid of the username dependency.

Reply:
I'd favor "something simple", at least as a fall-back when no user name is set.  Depending on a user name which isn't really required doesn't feel right and I can imagine at least one scenario where it may break:  On several sites, I've seen my password manager trying to fill out arbitrary fields that want a password.  If you're not careful, one might get set by accident.  If now the user name field stayed blank, that wouldn't actually hurt.  But if you need to have a user name for using WebAuthn (which it doesn't even require), it's just one tiny step closer to accidentally enabling password auth unintentionally.

Reply:
Ok, done in aaeaf0528035f29dfa24112c636b749d0818c362:

- Username config is no longer needed for WebAuthn registration.
- WebAuthn registration now sets `user.name` and `user.displayName` to the same value as follows:
  - If GUI username is set, use that. (This uses the _currently saved_ username, not necessarily the one in the username input field if that value has not yet been saved.)
  - Otherwise, if the server's device has a `name` set, use "Syncthing @ <deviceName>".
  - Otherwise, user "Syncthing @ <short device ID>".
  - If even that fails somehow, just use "Syncthing".

Does that sound reasonable?

Reply:
> I just noticed this won't work when generating an initial configuration. No idea currently how we can work around that.
> 
> This is what the settings GUI showed after just deleting config.xml:

Is this problem still present? I wasn't able to reproduce this behaviour on commit e179dd3b377cba1060c8780a9eb8515daa5c8dc4.

Reply:
That should have been fixed by https://github.com/emlun/syncthing/pull/11 I believe.

Reply:
I think this is resolved now?